Build failure due to missing include

When building OpenCL examples, I encountered the error:

```
vexcl/vexcl/backend/jit/device_vector.hpp:102:18: error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?
  102 |             std::copy_n(buffer.get<T>() + offset, size, host);
      |                  ^~~~~~
      |                  copy
```

Funciton `std::copy_n` is defined in `<algorithm>`, but the file `jit/device_vector.hpp` does not include `<algorithm>`. It looks like on some older compiler versions you would get `copy_n` as a transitive include through other headers, but not on GCC 14.2.0 that I am using.

Adding `#include <algorithm>` fixes compilation.

--- a/vexcl/backend/jit/device_vector.hpp
+++ b/vexcl/backend/jit/device_vector.hpp
@@ -31,8 +31,9 @@ THE SOFTWARE.
  * \brief  Wrapper around std::vector for the JIT backend.
  */
 
-#include <vector>
+#include <algorithm>
 #include <memory>
+#include <vector>
 
 namespace vex {
 namespace backend {

